uint16 width
uint16 height
float32[] data